org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -601,10 +603,36 @@ private static boolean isJavaIdentifierOrPeriod(char ch) {
 		return Character.isJavaIdentifierPart(ch) || ch == '.';
 	}
 
+	public static boolean isFolder(String uriString) {
+		IFile fakeFile = findFile(uriString); // This may return IFile even when uriString really describes a IContainer
+		IContainer parent = fakeFile == null ? null : fakeFile.getParent();
+		if (parent == null) {
+			return false;
+		}
+		String name = fakeFile.getName();
+		try {
+			if (!parent.isSynchronized(DEPTH_ONE)) {
+				parent.refreshLocal(DEPTH_ONE, null);
+			}
+			for (IResource member : parent.members()) {

```

Comment:
return parent.findMember(name) instanceof IFolder;

Reply:
agreed. that's a way better. fixed